components/http_client.rst

```diff
@@ -303,6 +333,29 @@ response sequentially instead of waiting for the entire response::
         fwrite($fileHandler, $chunk->getContent());
     }
 
+Canceling Responses
+~~~~~~~~~~~~~~~~~~~
+
+To abort a request (e.g. because it didn't complete in due time, or you want to
+fetch only the first bytes of the response, etc.), you can either:
+
+* use the ``cancel()`` method of ``ResponseInterface``::
+
+      $response->cancel()
+
+* throw an exception from a progress callback::
+
+      $response = $client->request('GET', 'https://..;', [
+        'on_progress' => function (int $dlNow, int $dlSize, array $info): void {
```

Comment:
Can we rename dlName and dlSize? What does Now mean, size is more clear but.. hmm 🧐😬

Reply:
that's the name in the code :)
dlNow => downloaded at the very moment this is called
dlSize => the total size of the content (-1 if unkown)
"dl" by opposition with "ul" for upload, since this is called also during upload (then $info contains more details)

Reply:
I'm going to move this in a dedicated section

Reply:
pushed